### PR TITLE
Support for DELETE and LIST (authenticated GET)

### LINKED ISSUE #2: Support vhl.ink methods
# Description
Currently, this only handles POST requests to make a new shortened link. Other methods such as getting the list of links and deleting are not yet supported.

# Relevant Information
[./src/handlers/shorten.ts] has the POST, other handlers can be added here to support command options, or entirely new slash commands could be added. A button or small interface would also be an interesting option.

### PR DESCRIPTION
**Pull request by nkalupahana**
Fixes #2. Support for:
* `/shorten`: will list all links
* `/shorten delete <path> <key>`: will delete link at path

### CODE REVIEW

**Review by SamLee514 — APPROVED** [review]
spicy work nisalsa

File: src/handlers/deletelink.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	try {
+		// https://api.slack.com/tutorials/slash-block-kit is more updated than the Cloudflare tutorial
+		const params: { [key: string]: string } | undefined = parseDeleteString(
+			text,
+		)?.groups;
+		const deleteKW: string | undefined = params?.delete;
+		const path: string | undefined = params?.path;
+		const key: string | undefined = params?.key;
+
+		if (deleteKW && path && key && deleteKW == "delete" && key == SECRET_KEY) {
```
Comment (aadibajpai) [inline src/handlers/deletelink.ts]: is it possible to consistently use ===

File: src/index.ts [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+			throw new Error('Request did not come from Slack.');
+		}
+
+		const formData = await request.formData();
+		const text = formData.get('text');
+
+		if (typeof text === 'string') {
+			const args = text.split(" ");
+
+			if (args.length == 1 && args[0] == "") {
```
Comment (aadibajpai) [inline src/index.ts]: and here

File: src/index.ts [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
+				return list();
+			} else if (args.length == 2) {
```
Comment (aadibajpai) [inline src/index.ts]: just everywhere

File: src/utils/index.ts [hunk 4]
```diff
@@ -58,6 +57,16 @@ export const parseShortenString = (text: string): RegExpMatchArray | null => {
 	return text.trim().match(shortenRegex);
 };
 
+const deleteRegex = /(?<delete>[\w\d-]+)\s+(?<path>[\S]+)\s+(?<key>[\S]+)/;
```
Comment (aadibajpai) [inline src/utils/index.ts:62]: a comment explaining the regex would be very useful for readability

File: src/index.ts (hunk 2 above)
Comment (aadibajpai) [reply src/index.ts]: might be nice to !args[0]

File: src/handlers/deletelink.ts (hunk 1 above)
Comment (aadibajpai) [reply src/handlers/deletelink.ts]: also if `deleteKW` is undefined then wouldn't `deleteKW === "delete"` automatically be false in that case? all that is to say do we need that extra first check

File: src/handlers/deletelink.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	});
+
+	return fetch(`${fetchUrl}`, {
+		method: 'DELETE',
+		headers
+	});
+};
+
+/**
+ * Handles shortening links. Expects the text to look like:
```
Comment (gfting) [inline src/handlers/deletelink.ts]: The comment should be adjusted to reflect what the function does, but looks good!

File: src/handlers/list.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import {
+	constructSlackMessage,
+} from '../utils';
+
+/**
+ * Sends POST to link shortening service
```
Comment (gfting) [inline src/handlers/list.ts]: Similarly, just adjust the comment here

File: src/handlers/list.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	});
+
+	return fetch(`${fetchUrl}`, {
+		method: 'GET',
+		headers
+	});
+};
+
+/**
+ * Handles shortening links. Expects the text to look like:
```
Comment (gfting) [inline src/handlers/list.ts]: Same here

File: src/index.ts (hunk 2 above)
Comment (gfting) [inline src/index.ts]: Could there also be a quick comment for what the arguments are supposed to be in this method, or documented as a tooltip somewhere? I noticed it below, but I think a general note in the README would be useful.

**Review by gfting — COMMENTED** [review]
Looks great overall, just a few minor comments based around documentation.

**Review by aadibajpai — APPROVED** [review]
ok hope this works

**Pull request merged**